node/actors/bft/src/leader/replica_prepare.rs

```diff
@@ -191,83 +171,19 @@ impl StateMachine {
             .into_values()
             .collect();
 
-        debug_assert!(num_messages == consensus.threshold());
-
-        // Get the highest block voted for and check if there's a quorum of votes for it. To have a quorum
-        // in this situation, we require 2*f+1 votes, where f is the maximum number of faulty replicas.
-        let mut count: HashMap<_, usize> = HashMap::new();
-
-        for vote in replica_messages.iter() {
-            *count.entry(vote.msg.high_vote.proposal).or_default() += 1;
-        }
-
-        let highest_vote: Option<validator::BlockHeader> = count
-            .iter()
-            // We only take one value from the iterator because there can only be at most one block with a quorum of 2f+1 votes.
-            .find(|(_, v)| **v > 2 * consensus.faulty_replicas())
-            .map(|(h, _)| h)
-            .cloned();
-
-        // Get the highest CommitQC.
-        let highest_qc: &validator::CommitQC = replica_messages
-            .iter()
-            .map(|s| &s.msg.high_qc)
-            .max_by_key(|qc| qc.message.view)
-            .unwrap();
-
-        // Create the block proposal to send to the replicas,
-        // and the commit vote to store in our block proposal cache.
-        let (proposal, payload) = match highest_vote {
-            // The previous block was not finalized, so we need to propose it again.
-            // For this we only need the header, since we are guaranteed that at least
-            // f+1 honest replicas have the block can broadcast when finalized
-            // (2f+1 have stated that they voted for the block, at most f are malicious).
-            Some(proposal) if proposal != highest_qc.message.proposal => (proposal, None),
-            // The previous block was finalized, so we can propose a new block.
-            _ => {
-                let payload = self
-                    .payload_source
-                    .propose(ctx, highest_qc.message.proposal.number.next())
-                    .await?;
-                metrics::METRICS
-                    .leader_proposal_payload_size
-                    .observe(payload.0.len());
-                let proposal =
-                    validator::BlockHeader::new(&highest_qc.message.proposal, payload.hash());
-                (proposal, Some(payload))
-            }
-        };
+        debug_assert!(num_messages == self.inner.threshold());
```

Comment:
Nit: You can use `debug_assert_eq!` here.

Reply:
done